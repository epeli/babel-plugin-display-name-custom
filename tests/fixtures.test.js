const babel = require("babel-core");
const fs = require("fs");

const fixtures = fs.readdirSync(__dirname + "/fixtures");


describe("compile fixtures to snapshots:", () => {

    fixtures.forEach(fixtureFile => {

        test(fixtureFile, () => {

            const filePath = __dirname + "/fixtures/" + fixtureFile;

            const lines = fs.readFileSync(filePath).toString().split("\n");

            const options = JSON.parse(lines[0].slice(2).trim());

            const output = babel.transformFileSync(filePath, {
                babelrc: false,
                plugins: [
                    [__dirname + "/../plugin.js", options],
                ],
            });

            expect(output.code).toMatchSnapshot();
        });

    });

});
