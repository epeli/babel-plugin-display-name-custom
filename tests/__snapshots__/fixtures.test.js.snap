exports[`alternative-name.js 1`] = `
"// {\"modules\": {\"react-simple\": {\"default\": true}}}
import create from \"react-simple\";

const Foo = create(\"div\", {
    color: \"red\"
});
Foo.displayName = \"Foo\";"
`;

exports[`argument-shadow.js 1`] = `
"// {\"react-simple\": {\"default\": true}}
import simple from \"react-simple\";

function fn(simple) {
    var no = simple();
}

function outer(simple) {
    function inner() {
        var no = simple();
    }
}"
`;

exports[`assignment-only.js 1`] = `
"// {\"react-simple\": {\"default\": true}}
import simple from \"react-simple\";

var no = simple;"
`;

exports[`basic.js 1`] = `
"// {\"modules\": {\"react-simple\": {\"default\": true}}}
import simple from \"react-simple\";

const Foo = simple(\"div\", {
    color: \"red\"
});
Foo.displayName = \"Foo\";"
`;

exports[`custom.js 1`] = `
"// {\"modules\": {\"other\": {\"default\": true}}}
import createComponent from \"other\";

const Foo = createComponent(\"div\", {
    color: \"red\"
});
Foo.displayName = \"Foo\";"
`;

exports[`declaration-shadow.js 1`] = `
"// {\"react-simple\": {\"default\": true}}
import simple from \"react-simple\";

function fn() {
    const simple = () => null;
    var no = simple();
}

function outer() {
    const simple = () => null;
    function inner() {
        var no = simple();
    }
}"
`;

exports[`export.js 1`] = `
"// {\"modules\": {\"react-simple\": {\"default\": true}}}
import simple from \"react-simple\";

export const Foo = simple(\"div\", {
    color: \"red\"
});
Foo.displayName = \"Foo\";"
`;

exports[`inside-function.js 1`] = `
"// {\"modules\": {\"react-simple\": {\"default\": true}}}
import simple from \"react-simple\";

function fn() {
    const Foo = simple(\"div\", {
        color: \"red\"
    });
    Foo.displayName = \"Foo\";
}

function outer() {
    function inner() {
        const Foo = simple(\"div\", {
            color: \"red\"
        });
        Foo.displayName = \"Foo\";
    }
}"
`;

exports[`local-default-export.js 1`] = `
"// {\"modules\": {\"./tests/fixtures/local\": {\"default\": true}}}
import createComponent, { no } from \"./local\";

const Foo = createComponent(\"div\", {
    color: \"red\"
});

Foo.displayName = \"Foo\";
var nope = no();"
`;

exports[`local-named-export.js 1`] = `
"// {\"modules\": {\"./tests/fixtures/local\": {\"createComponent\": true}}}
import { createComponent as other, no } from \"./local\";

const Foo = other(\"div\", {
    color: \"red\"
});

Foo.displayName = \"Foo\";
var nope = no();"
`;

exports[`named-export.js 1`] = `
"// {\"modules\": {\"amodule\": {\"createComponent\": true}}}
import noDefault, { createComponent, no } from \"amodule\";

const Foo = createComponent(\"div\", {
    color: \"red\"
});

Foo.displayName = \"Foo\";
var nopeDefault = noDefault();
var nope = no();"
`;

exports[`named-export-as.js 1`] = `
"// {\"modules\": {\"amodule\": {\"createComponent\": true}}}
import { createComponent as other, no } from \"amodule\";

const Foo = other(\"div\", {
    color: \"red\"
});

Foo.displayName = \"Foo\";
var nope = no();"
`;
